packages/react/src/enhancers/withServerPlaceholder.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { ComponentType } from 'react';
+import { ServerPlaceholder } from '../components/Placeholder';
+import { ComponentRendering } from '@sitecore-content-sdk/core/layout';
+import { ComponentMap } from '../components/sharedTypes';
+import { Page } from '@sitecore-content-sdk/core/client';
+
+export type ComponentProps = {
+  rendering: ComponentRendering;
+  placeholders: Record<string, React.ReactNode>;
+  componentMap?: ComponentMap;
+};
+
+export type WrapperProps = {
+  page: Page;
+  componentMap?: ComponentMap;
+  rendering: ComponentRendering;
+};
+
+export const withServerPlaceholder = <T extends ComponentProps, W extends T & WrapperProps>(
```

Comment:
i think we should consider adding displayName to the returned component for better DevTools developer experience
  const wrappedName = Component.displayName || Component.name || 'Component';
 ...
   Wrapped.displayName = `withServerPlaceholder(${wrappedName})`;
   return Wrapped;